src/rgw/rgw_rados.cc

```diff
@@ -4478,6 +4472,8 @@ int RGWRados::copy_obj(RGWObjectCtx& obj_ctx,
     append_rand_alpha(cct, tag, tag, 32);
   }
 
+  auto aio = rgw::make_throttle(cct->_conf->rgw_max_copy_obj_concurrent_io, y);
+  rgw::AioResultList all_results;
```

Comment:
it looks like we could avoid allocating `aio` for the `copy_itself` case?
```c++
  std::unique_ptr<rgw::Aio> aio;
  rgw::AioResultList all_results;
  if (!copy_itself) {
    aio = rgw::make_throttle(cct->_conf->rgw_max_copy_obj_concurrent_io, y);
```

Reply:
Well.